### PR TITLE
Use Threads inside of Partitions instead of Processes

### PR DESCRIPTION
**Pull request by sevenautumns**

### CODE REVIEW

**Comment by florianhartung** [comment]
Note: this should also close https://github.com/DLR-FT/a653rs-linux/issues/85

File: hypervisor/src/hypervisor/partition.rs [hunk 1]
```diff
@@ -205,6 +216,14 @@ impl Run {
                         None,
                     )
                     .unwrap(),
+                    FileMounter::new(
+                        Some(ipc_path),
+                        PartitionConstants::IPC_SENDER[1..].into(),
```
Comment (dadada) [inline hypervisor/src/hypervisor/partition.rs:221]: Use named variable instead. See line 104.

File: partition/Cargo.toml [hunk 2]
```diff
@@ -22,5 +22,5 @@ once_cell.workspace = true
 lazy_static = "1.4"
 anyhow = "1.0"
 log.workspace = true
-safemem = "0.3"
 tinyvec = "1.6"
+oneshot = "0.1.6"
```
Comment (dadada) [inline partition/Cargo.toml:26]: Newline at EOF

File: partition/src/lib.rs [hunk 3]
```diff
@@ -99,14 +79,6 @@ pub(crate) static UDP_IO_RX: Lazy<IoReceiver<UdpSocket>> =
 pub(crate) static TCP_IO_RX: Lazy<IoReceiver<TcpStream>> =
     Lazy::new(|| unsafe { IoReceiver::<TcpStream>::from_raw_fd(CONSTANTS.io_fd) });
```
Comment (dadada) [inline partition/src/lib.rs:81]: Informational: It is a shame the socket needs to be connected in the hypervisor. We may think about using something like [slirp](https://wiki.qemu.org/Documentation/Networking) for more flexible networking.

**Review by dadada — APPROVED** [review]

**Review by dadada — APPROVED** [review]

**Pull request merged**